### ISSUE TITLE
Question about compute_train_steps in best practice

### ISSUE THREAD

**Issue by lovecambi** [ByteDance-Seed/VeOmni#116]
The note of [best practice](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs/start/best_practice.md#dataset)  says
```
if you dataset is mapping, you are recommended to add pass the len(train_dataset) to the train_steps to compute the correct train steps.
```

However, according to the implementation of [compute_train_steps](https://github.com/ByteDance-Seed/VeOmni/blob/main/veomni/utils/arguments.py#L650), if `rmpad` or `rmpad_with_pos_ids` is True, the `len(train_dataset)` won't be used to calculate train steps. Or, does it mean if the dataset is mapping, it is recommended to set both `rmpad` and `rmpad_with_pos_ids` as False?

**Comment by Coach257** [comment]
Using rmpad or rmpad_with_pos_ids is also acceptable when using mapping dataset. You should precompute the `train_size` and compute the correct train steps. 
Check [compute_train_steps](https://github.com/ByteDance-Seed/VeOmni/blob/74ca8d72e27511d8bc1b236076fd7fe1d28f7288/veomni/utils/arguments.py#L691C13-L691C111)
